code/modules/organs/cybernetics/upgrades/adapters.dm

```diff
@@ -0,0 +1,25 @@
+//Enhancers are used to bring a cybernetic organ to a higher tier
+
+/obj/item/cybernetics/enhancer
+	name = "cybernetic limb enhancer"
+	var/tier = 1 //What tier the enhancer will work on
+
+/obj/item/cybernetics/enhancer/New()
+	..()
+	name = "[name] Tier [tier]"
+
+/obj/item/cybernetics/enhancer/apply(var/obj/item/O)
+	if(istype(O, obj/item/organ))
+		var/currentTier = O.organTier
+		if(currentTier >= tier)
+			to_chat(usr, "\the [src]'s tier is lower or equal that of \the [O]'s tier.")
+			return
+		else if(tier = currentTier + 1)
+			//This is so hacky
+			switch(currentTier)
```

Comment:
About a quarter done, was intended to cycle through the tiers and up the module amount that way, haven't done the variables and how it's approached on the limb yet though.